storage-node-v2/src/services/webApi/controllers/publicApi.ts

```diff
@@ -363,3 +352,29 @@ function sendResponseWithError(res: express.Response, err: Error, errorType: str
 function isNofileError(err: Error): boolean {
   return err.toString().includes('ENOENT')
 }
+
+/**
+ * Get the status code by error.
+ *
+ * @param err - error
+ * @returns HTTP status code
+ */
+function getHttpStatusCodeByError(err: Error): number {
+  if (isNofileError(err)) {
+    return 404
+  }
+
+  if (err instanceof ExtrinsicFailedError) {
+    return 400
+  }
+
+  if (err instanceof WebApiError) {
+    return 400
+  }
+
+  if (err instanceof CLIError) {
+    return 400
+  }
+
+  return 500
```

Comment:
I think it's good that there are no `410`s anymore, but:
- `openapi` spec still includes `410` response for `uploadFile`
- `ExtrinsicFailedError`, `WebApiError` and `CLIError` are not necessarly all `400`, a few examples of how I think they should be categorized:
  - `Max file size exceeded.` is a `400`, because the request is invalid (it's obviously client-side error)
  - `Invalid upload token request signature.` / `Provided controller account and member id don't match.` can be `400` or `401` (more specifically), as both indicate the request was not valid (or more specifically - that authroization was unsuccesful)
  - `No Joystream API loaded.` / `No Joystream account loaded.` / `No upload directory path loaded.` are `500` or at least in `5xx` category, as those are server (node) issues, not client-side errors (caused by invalid requests).

Reply:
Fixed in 84c14da